network/builder/src/builder.rs

```diff
@@ -302,25 +306,42 @@ impl NetworkBuilder {
             &counters::PENDING_CONNECTIVITY_MANAGER_REQUESTS,
         );
         self.conn_mgr_reqs_tx = Some(conn_mgr_reqs_tx);
-        let trusted_peers = self.trusted_peers.clone();
-        let seed_peers = self.seed_peers.clone();
-        let max_connection_delay_ms = self.max_connection_delay_ms;
-        let connectivity_check_interval_ms = self.connectivity_check_interval_ms;
         let pm_conn_mgr_notifs_rx = self.add_connection_event_listener();
-        let conn_mgr = self.executor.enter(|| {
-            ConnectivityManager::new(
-                self.network_context.clone(),
-                trusted_peers,
-                seed_peers,
-                interval(Duration::from_millis(connectivity_check_interval_ms)).fuse(),
-                ConnectionRequestSender::new(self.connection_reqs_tx.clone()),
-                pm_conn_mgr_notifs_rx,
-                conn_mgr_reqs_rx,
-                ExponentialBackoff::from_millis(2).factor(1000),
-                max_connection_delay_ms,
-            )
-        });
-        self.executor.spawn(conn_mgr.start());
+
+        let config = ConnectivityManagerBuilderConfig::new(
+            self.network_context().clone(),
+            self.trusted_peers.clone(),
+            self.seed_peers.clone(),
+            self.connectivity_check_interval_ms,
+            2, // Hard-coded constant
```

Comment:
either don't expose this or make it a const somewhere

Reply:
Actually, It should go to a config somewhere.    
But yes.

(too many magic constants)